src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidAccountManagerBroker.cs

```diff
@@ -0,0 +1,455 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Android.App;
+using Android.Content;
+using AndroidNative = Android;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal.Broker;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.UI;
+using Microsoft.Identity.Json.Linq;
+using Microsoft.Identity.Client.Internal.Requests;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Http;
+using System.Net;
+using Android.OS;
+using System.Linq;
+using Android.Accounts;
+using Java.Util.Concurrent;
+using OperationCanceledException = Android.Accounts.OperationCanceledException;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.Identity.Client.Cache;
+using Microsoft.Identity.Client.Instance.Discovery;
+
+namespace Microsoft.Identity.Client.Platforms.Android.Broker
+{
+    [AndroidNative.Runtime.Preserve(AllMembers = true)]
+    internal class AndroidAccountManagerBroker : IBroker
+    {
+        private long AccountManagerTimeoutSeconds { get; } = 5 * 60;
+        //Since the correlation ID is not returned from the broker response, it must be stored at the beginning of the authentication call and re-injected into the response at the end.
+        private static string s_correlationId;
+        private readonly AndroidBrokerHelper _brokerHelper;
+        private readonly ICoreLogger _logger;
+        private readonly Activity _parentActivity;
+
+        public AndroidAccountManagerBroker(CoreUIParent uiParent, ICoreLogger logger)
+        {
+            _parentActivity = uiParent?.Activity;
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            AuthenticationContinuationHelper.LastRequestLogger = _logger;
+            _brokerHelper = new AndroidBrokerHelper(Application.Context, logger);
+        }
+
+        public bool IsBrokerInstalledAndInvokable()
+        {
+            return _brokerHelper.IsBrokerInstalledAndInvokable();
+        }
+
+        public async Task<MsalTokenResponse> AcquireTokenInteractiveAsync(
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenInteractiveParameters acquireTokenInteractiveParameters)
+        {
+
+            CheckPowerOptimizationStatus();
+
+            AndroidBrokerStaticHelper.InteractiveBrokerTokenResponse = null;
+
+            BrokerRequest brokerRequest = BrokerRequest.FromInteractiveParameters(
+                authenticationRequestParameters, acquireTokenInteractiveParameters);
+
+            // There can only be 1 broker request at a time so keep track of the correlation id
+            s_correlationId = brokerRequest.CorrelationId;
```

Comment:
>s_correlationId [](start = 12, length = 15)

how is this keeping track of the request? i don't see it used in the class after being set here.

Reply:
this is using the wrong variable, will update